File: Week 1 - C/credit.c
#include <stdio.h>
#include <stdbool.h>
#include <cs50.h>
#include <string.h>
#include <stdlib.h>

int main()
{
    long credit = get_long("Input credit number here: ");
    //4003600000000014
    char credit_string[100];
    sprintf(credit_string, "%li", credit);
    int length1;
    length1 = strlen(credit_string);

    int i = 0;
    int sum1 = 0;
    for (i = 0; credit_string[i] != '\0'; i ++)
    {
        if (length1 % 2 == 0)
        {
            if (i%2==0)
            {
            int temp1 = (credit_string[i] - 48) * 2;

            bool x1 = false;
            while (x1==false)
                {
                sum1 += temp1 % 10;
                temp1 = temp1 / 10;
                if (temp1 < 1)
                    {
                    break;
                    }
                }
            }
        }
        else
        {
            if (i%2!=0)
            {
            int temp1 = (credit_string[i] - 48) * 2;

            bool x1 = false;
            while (x1==false)
                {
                sum1 += temp1 % 10;
                temp1 = temp1 / 10;
                if (temp1 < 1)
                    {
                    break;
                    }
                }
            }
        }
    }

    int i1 =0;
    int sum2 = 0;
    for (i1 = 0; credit_string[i1] != '\0'; i1 ++)
    {
        if (length1 % 2 == 0)
        {
            if (i1 % 2!=0)
        {
            int temp1 = credit_string[i1] - 48;
            sum1 = sum1 + temp1;
        }
        }
        else
        {
            if (i1 % 2==0)
        {
            int temp1 = credit_string[i1] - 48;
            sum1 = sum1 + temp1;
        }
        }

    }

    char finalstr[20];
    sprintf(finalstr,"%d",sum1);
    int length_final = strlen(finalstr);

    int last = finalstr[length_final-1]-48;


    int master = (credit_string[0] -48) * 10 + (credit_string[1]-48);


    if (last == 0)
    {
        if (credit_string[0] == '4')
        {
            if (length1 == 13 || length1 == 16)
            {
                printf("VISA\n");
            }
            else
            {
                printf("INVALID\n");
            }
        }
        else if (master >= 51 && master <= 55)
        {
            if (length1 == 16)
            {
                printf("MASTERCARD\n");
            }
            else
            {
                printf("INVALID\n");
            }
        }
        else if (master == 34 || master == 37)
        {
            if (length1 == 15)
            {
                printf("AMEX\n");
            }
            else
            {
                printf("INVALID\n");
            }
        }
        else
        {
            printf("INVALID\n");
        }
    }
    else
    {
        printf("INVALID\n");
    }
}

File: Week 6 - Python/credit.py
import cs50

n1 = cs50.get_int("Input credit card number")

str1 = str(n1)[::-1]
count1 = 0
count2 = 0
for i in range(len(str1)):
    if i % 2 != 0:
        temp1  = int(str1[i]) * 2
        b1 = False
        while b1==False:
            count1 += temp1 % 10
            temp1 = temp1 // 10
            if temp1 <= 0:
                break
    else:
        count2 += int(str1[i])

final = count1 + count2
length = len(str1)
start = int(str(n1)[0])*10 + int(str(n1)[1])
if final % 10 != 0:
    print("INVALID")
else:
    flag = 0
    if length == 15:
        if start == 34 or start == 37:
            flag+=1
            print("AMEX")
    if length == 16:
        if start >= 51 and start <=55:
            flag+=1
            print("MASTERCARD")
    if length == 13 or length == 16:
        if int(str1[-1]) == 4:
            flag+=1
            print("VISA")
    if flag == 0:
        print("INVALID")
